bin/millau/runtime/src/rialto_messages.rs

```diff
@@ -87,9 +88,16 @@ impl MessageBridge for WithRialtoMessageBridge {
 	type ThisChain = Millau;
 	type BridgedChain = Rialto;
 
-	fn maximal_dispatch_weight_of_message_on_bridged_chain() -> Weight {
+	fn weight_limits_of_message_on_bridged_chain(message_payload: &[u8]) -> RangeInclusive<Weight> {
 		// we don't want to relay too large messages + keep reserve for future upgrades
-		bp_rialto::MAXIMUM_EXTRINSIC_WEIGHT / 2
+		let upper_limit = bp_rialto::MAXIMUM_EXTRINSIC_WEIGHT / 2;
+
+		// given Rialto chain parameters (`TransactionByteFee`, `WeightToFee`, `FeeMultiplierUpdate`),
+		// the minimal weight of the message may be computed as message.length() + minimal extrinsic weight
+		let message_payload_length_weight = Weight::try_from(message_payload.len()).unwrap_or(Weight::MAX);
```

Comment:
`message_payload` is just `Call`, right? So it doesn't include the transaction overhead (signed extensions, signatures, etc), I think we should add some extra constant to cater for this.
Instead of using `BASE_EXTRINSIC_WEIGHT` I would use something like: the weight of transaction with `System::remark(vec![])` call and have it measured externally and hardcoded.

Reply:
Right, that's the encoded `Call`. Signed extensions and signatures checks are not a part of the call dispatch, right? So `Call::dispatch()` will not verify any signatures or signed extensions.

Verification will be done by the delivery transaction itself, which will hand the encoded call to the call-dispatch module, which in turn would verify signatures (if required). So signatures verification weight is included into delivery transaction base weight (which is accounted in the dispatch fee through the `weight_of_delivery_transaction()` call).

So here we deal only with pure call weight. Including overhead here doesn't make sense, because it is already accounted.

Reply:
Thanks for explanations. I thought this should be the minimal weight of the entire transaction.
We've also discussed off-band that the `BASE_EXTRINSIC_WEIGHT` doesn't need to be added here, since it's rather part of `weight_of_delivery_transaction()`, since it's payed per-extrinsic and not per-dispatch, so each message only requires a minimal weight to cover the length-weight cost.